pkg/volume/volume.go

```diff
@@ -68,6 +69,37 @@ func NewVolumeOperation(volume *v1alpha1.CASVolume) (*VolumeOperation, error) {
 	}, nil
 }
 
+// getCloneLabels returns a map of clone specific configuration
+func (v *VolumeOperation) getCloneLabels() (map[string]interface{}, error) {
+	// Initially all the values are set to their defaults
+	cloneLabels := map[string]interface{}{
+		string(v1alpha1.SnapshotNameVTP):         "",
+		string(v1alpha1.SourceVolumeTargetIPVTP): "",
+		string(v1alpha1.IsCloneEnableVTP):        "false",
+		string(v1alpha1.StorageClassVTP):         "",
+		string(v1alpha1.SourceVolumeVTP):         "",
+	}
+
+	// if volume is clone enabled then update cloneLabels map
+	if v.volume.CloneSpec != nil {
```

Comment:
If we avoid pointer then this condition will change.

In addition I guess this is the fact that changes the flow from volume to clone or vice versa. Can we move this condition to higher up layer i.e. at some caller code and avoid modifying this code. The cost of violating open closed principle is much more here in the entire workflow of creating a clone. It will be cheaper to have this condition at a higher up layer.

Reply:
Changed to inline object instead of pointer.